src/abstractarray.jl

```diff
@@ -161,6 +161,8 @@ reshape(a::Array, s::Size{S}) where {S} = s(a)
 
 @inline copy(a::StaticArray) = typeof(a)(Tuple(a))
 
+@inline reverse(v::SVector) = SVector(reverse(Tuple(v)))
```

Comment:
Ideally. this should be generic with respect to type and dimensionality.

Maybe like `@inline reverse(a::StaticArray) = typeof(a)(reverse(Tuple(a)))` (or else via `similar_type`).

Reply:
Not sure why you want this extended to all `StaticArrays` with dimension above 1, since `reverse` does not work (without keyword arguments) for matrices etc. Eg
```julia
julia> reverse(ones(2, 2))
ERROR: UndefKeywordError: keyword argument dims not assigned
Stacktrace:
 [1] reverse(::Array{Float64,2}) at ./arraymath.jl:62
 [2] top-level scope at none:0
```

Reply:
Sometimes `Base` gets it wrong? ¯\_(ツ)_/¯

`@inline reverse(a::StativVector) = typeof(a)(reverse(Tuple(a)))` is OK if you feel that is safer.

Reply:
`v = MVector(1, 2, 3)`. The result should be an `MVector` rather than `SVector`.

Reply:
But `reverse(::MVector)` works fine already. I am just adding a method `reverse(::SVector)`.

Reply:
Consider also other non-`SVector` immutable `StaticVector`s:

```julia
julia> struct Point3D <: FieldVector{3, Float64}
           x::Float64
           y::Float64
           z::Float64
       end

julia> reverse(Point3D(1, 2, 3))
3-element MArray{Tuple{3},Float64,1,3}:
 3.0
 2.0
 1.0
```

Also, widening to `StaticVector` improves performance for `MVector`:

```julia
julia> @btime reverse(x) setup = x = MVector(1, 2, 3)
  9.667 ns (1 allocation: 32 bytes)

julia> @inline Base.reverse(v::StaticVector) = typeof(v)(reverse(Tuple(v)))

julia> @btime reverse(x) setup = x = MVector(1, 2, 3);
  5.777 ns (1 allocation: 32 bytes)
```

Lastly, for consistency, the only other place where `SVector` is special-cased like this is

https://github.com/JuliaArrays/StaticArrays.jl/blob/ae146942c54bac73859dca828278b356130e3f8f/src/SDiagonal.jl#L7

and I consider that to be an oversight.

Reply:
All those things would be nice I guess, but this is a PR for fixing #566. I don't understand why you are asking me to do other things outside the scope of this PR in a code review.

Reply:
To understand, you need to think like a maintainer for a moment:

Every line of code is a tradeoff between
* The functionality it brings (valuable and visible to everyone)
* The maintenance burden it introduces (largely invisible to users, except that difficult to maintain packages have more bugs)

When you introduce code which only works in one specific case, this tradeoff doesn't look so good to from the maintainers point of view, and it might be rejected. This depends a lot on how bad the problem is and how hard it is to fix in general. (In this case, not so bad, and very easy to fix, respectively.)